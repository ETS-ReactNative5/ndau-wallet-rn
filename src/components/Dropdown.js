import React, { Component } from 'react';

import { StyleSheet } from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp
} from 'react-native-responsive-screen';
import ModalDropdown from 'react-native-modal-dropdown';

class CommonButton extends Component {
  onPress() {
    this.props.onPress();
  }
  render() {
    return (
      <ModalDropdown
        defaultValue={this.props.defaultValue}
        // defaultIndex={0}
        options={this.props.options}
        onSelect={this.props.onSelect}
        style={styles.containerStyle}
        textStyle={styles.text}
        dropdownStyle={styles.dropdownContainerStyle}
        dropdownTextStyle={styles.text}
      />
    );
  }
}

var styles = StyleSheet.create({
  text: {
    color: '#000000',
    fontSize: 16,
    fontFamily: 'TitilliumWeb-Regular'
  },
  containerStyle: {
    alignContent: 'center',
    justifyContent: 'center',
    borderRadius: 3,
    height: hp('7%'),
    width: wp('90%'),
    backgroundColor: '#ffffff',
    paddingLeft: wp('1%')
  },
  dropdownContainerStyle: {
    // alignContent: 'center',
    // justifyContent: 'center',
    borderRadius: 3,
    // height: hp('7%'),
    width: wp('88%'),
    backgroundColor: '#ffffff',
    // marginRight: wp('30%'),
    marginTop: wp('3%')
  }
});

export default CommonButton;
