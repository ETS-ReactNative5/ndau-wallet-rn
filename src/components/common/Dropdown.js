import React, { Component } from 'react'

import { StyleSheet } from 'react-native'
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp
} from 'react-native-responsive-screen'
import ModalDropdown from 'react-native-modal-dropdown'

let full = false

class Dropdown extends Component {
  constructor (props) {
    super(props)

    if (this.props.full) {
      full = this.props.full
    }
  }
  onPress () {
    this.props.onPress()
  }
  render () {
    return (
      <ModalDropdown
        defaultValue={this.props.defaultValue}
        options={this.props.options}
        onSelect={this.props.onSelect}
        style={[styles.containerStyle, this.props.style]}
        textStyle={styles.text}
        dropdownStyle={styles.dropdownContainerStyle}
        dropdownTextStyle={styles.text}
      />
    )
  }
}

var styles = StyleSheet.create({
  text: {
    color: '#000000',
    fontSize: 18,
    fontFamily: 'TitilliumWeb-Regular'
  },
  containerStyle: {
    alignContent: 'center',
    justifyContent: 'center',
    borderRadius: 3,
    height: hp('7%'),
    width: wp('96%'),
    backgroundColor: '#293e63',
    paddingLeft: wp('1%')
  },
  dropdownContainerStyle: {
    borderRadius: 3,
    width: wp('94%'),
    backgroundColor: '#293e63',
    marginTop: wp('3%')
  }
})

export default Dropdown
