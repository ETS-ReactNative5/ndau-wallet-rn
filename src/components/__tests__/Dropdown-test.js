import { mount } from 'enzyme'
import React from 'react'
import { StyleSheet } from 'react-native'
import Dropdown from '../Dropdown'

import renderer from 'react-test-renderer'

describe('testing Dropdown...', () => {
  let styles = StyleSheet.create({
    wizardText: {
      color: '#ffffff',
      fontSize: 20
    }
  })
  const navigator = {
    setStyle: () => {}
  }

  it('renders correctly', () => {
    const tree = renderer
      .create(<Dropdown parentStyles={styles} navigator={navigator} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
