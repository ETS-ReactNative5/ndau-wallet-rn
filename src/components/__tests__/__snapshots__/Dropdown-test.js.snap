// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Dropdown... renders correctly 1`] = `
<View
  animated={true}
  defaultIndex={-1}
  defaultValue="Please select..."
  disabled={false}
  dropdownStyle={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 3,
      "marginTop": 22.5,
      "width": 705,
    }
  }
  dropdownTextStyle={
    Object {
      "color": "#000000",
      "fontFamily": "TitilliumWeb-Regular",
      "fontSize": 18,
    }
  }
  keyboardShouldPersistTaps="never"
  options={null}
  scrollEnabled={true}
  showsVerticalScrollIndicator={true}
  style={
    Array [
      Object {
        "alignContent": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 3,
        "height": 93.5,
        "justifyContent": "center",
        "paddingLeft": 7.5,
        "width": 750,
      },
      undefined,
    ]
  }
  textStyle={
    Object {
      "color": "#000000",
      "fontFamily": "TitilliumWeb-Regular",
      "fontSize": 18,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontSize": 12,
            },
            Object {
              "color": "#000000",
              "fontFamily": "TitilliumWeb-Regular",
              "fontSize": 18,
            },
          ]
        }
      >
        Please select...
      </Text>
    </View>
  </View>
</View>
`;
